spark-3/src/test/scala/org/neo4j/spark/DataSourceReaderTSE.scala

```diff
@@ -1649,6 +1648,65 @@ class DataSourceReaderTSE extends SparkConnectorScalaBaseTSE {
     assertEquals(8L, df.collect().head.get(0))
   }
 
+  @Test
+  def test531(): Unit = {
+    val dataFrame = ss.read.format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("query",
+        """
+          |UNWIND [{first: '2022-06-14T10:02:28.192Z', second: null},
+          | {first: '2022-06-15T10:02:28.192Z', second: '2022-06-16T10:02:28.192Z'}] AS event
+          |RETURN datetime(event.first) AS first, datetime(event.second) AS second
+          |""".stripMargin)
+      .load()
+
+    assertEquals(
+      StructType(Array(
+        StructField("first", DataTypes.TimestampType),
+        StructField("second", DataTypes.StringType)
+      )),
+      dataFrame.schema
+    )
+    assertEquals(
+      Seq(
+        (Timestamp.valueOf("2022-06-14 12:02:28.192"), null),
+        (Timestamp.valueOf("2022-06-15 12:02:28.192"), "2022-06-16T10:02:28.192Z")
+      ),
+      dataFrame.collect()
+        .map(r => (r.getTimestamp(0), r.getString(1)))
+        .toSeq
+    )
+
+    val schema = StructType(Array(
+      StructField("first", DataTypes.TimestampType),
+      StructField("second", DataTypes.TimestampType),
+    ))
+    // dataFrameWithSchema must with the proper data type
+    val dataFrameWithSchema = ss.read.format(classOf[DataSource].getName)
```

Comment:
can you maybe extract this part into a second test?

Reply:
Done